Formula/litani.rb

```diff
@@ -0,0 +1,69 @@
+class Litani < Formula
+  desc "Metabuild system"
+  homepage "https://awslabs.github.io/aws-build-accumulator/"
+  url "https://github.com/awslabs/aws-build-accumulator/archive/refs/tags/1.18.0.tar.gz"
+  sha256 "5a56c8675cfde39b7308eac78fd8d8af290d83e241063fb2d7fb9545062d6daf"
+  license "Apache-2.0"
+
+  depends_on "coreutils" => :build
+  depends_on "mandoc" => :build
+  depends_on "scdoc" => :build
+  depends_on "gnuplot"
+  depends_on "ninja"
+  depends_on "python@3.9"
+
+  resource "Jinja2" do
+    url "https://files.pythonhosted.org/packages/39/11/8076571afd97303dfeb6e466f27187ca4970918d4b36d5326725514d3ed3/Jinja2-3.0.1.tar.gz"
+    sha256 "703f484b47a6af502e743c9122595cc812b0271f661722403114f71a79d0f5a4"
+  end
+
+  resource "MarkupSafe" do
+    url "https://files.pythonhosted.org/packages/bf/10/ff66fea6d1788c458663a84d88787bae15d45daa16f6b3ef33322a51fc7e/MarkupSafe-2.0.1.tar.gz"
+    sha256 "594c67807fb16238b30c44bdf74f36c02cdf22d1c8cda91ef8a0ed8dabf5620a"
+  end
+
+  resource "voluptuous" do
+    url "https://files.pythonhosted.org/packages/c0/2c/ccbeb25364e3e0c5e4522f13d66e2fc639bb4d4ecdf73be0959552cbecb4/voluptuous-0.12.2.tar.gz"
+    sha256 "4db1ac5079db9249820d49c891cb4660a6f8cae350491210abce741fabf56513"
+  end
+
+  resource "graphviz" do
+    url "https://files.pythonhosted.org/packages/f6/14/aa3ec10e4ab1524ba69f4742ef9cfa01fd668d0840afe5d5e6f708a95031/graphviz-0.19.1.zip"
+    sha256 "09ed0cde452d015fe77c4845a210eb642f28d245f5bc250d4b97808cb8f49078"
+  end
+
+  resource "PyYAML" do
+    url "https://files.pythonhosted.org/packages/a0/a4/d63f2d7597e1a4b55aa3b4d6c5b029991d3b824b5bd331af8d4ab1ed687d/PyYAML-5.4.1.tar.gz"
+    sha256 "607774cbba28732bfa802b54baa7484215f530991055bb562efbed5b2f20a45e"
+  end
+
+  def install
+    xy = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
+    ENV.prepend_create_path "PYTHONPATH", lib/"python#{xy}/site-packages"
+    resources.each do |r|
+      r.stage do
+        system Formula["python@3.9"].opt_bin/"python3", *Language::Python.setup_install_args(prefix)
+      end
+    end
+    (libexec/"bin").install Dir["*"]
+    bin.install_symlink libexec/"bin/litani"
+
+    cd libexec/"bin/doc" do
+      system Formula["python@3.9"].opt_bin/"python3", "configure"
+      system "ninja"
+    end
+    man.install Dir[libexec/"bin/doc/out/man/*.roff"]
+    doc.install Dir[libexec/"bin/doc/out/html/index.html"]
```

Comment:
Actually, this might be simpler. Apologies for overwhelming you with suggestions. This third one is probably ideal.
```suggestion
    venv = virtualenv_create(libexec/"vendor", "python3")
    venv.pip_install resources

    libexec.install Dir["*"] - ["doc"]
    (bin/"litani").write_env_script libexec/"litani", PATH: "\"#{libexec}/vendor/bin:${PATH}\""

    cd "doc" do
      system libexec/"vendor/bin/python3", "configure"
      system "ninja", "--verbose"
    end

    man.install Pathname.glob("doc/out/man/*.roff")
    doc.install "doc/out/html/index.html"
```

Reply:
Thank you for all the suggestions and help. You have been really responsive and helpful.
I made the change you suggested and also added
```
ENV.prepend_create_path "PYTHONPATH", libexec/"vendor"/Language::Python.site_packages("python3")
````
as `system "ninja", "--verbose"` seems to be failing without it with a `jinja2 missing` error.

----

One question I have is if i update Python version from `@3.9` to `@3.10`, it seems to be failing with the `missing jinja2 error` as well. What should I change to make sure that doesn't happen.

----

Also seems like `gnuplot` has a dependency of python@3.9 which is conflicting if i install `python@3.10` along with it. How can i get past it?

Reply:
Instead of prepending to `PYTHONPATH`:
```ruby
ENV.prepend_path "PATH", libexec/"vendor/bin"
```
That way you're using your venv's interpreter.

Reply:
> Also seems like `gnuplot` has a dependency of python@3.9 which is conflicting if i install `python@3.10` along with it. How can i get past it?

You can either drop the `gnuplot` dependency, stay with `python@3.9`, or help migrate `gnuplot` to `python@3.10`. Migrating `gnuplot` may require migrating many other formulae.

Reply:
> Instead of prepending to `PYTHONPATH`:
> 
> ```ruby
> ENV.prepend_path "PATH", libexec/"vendor/bin"
> ```
> 
> That way you're using your venv's interpreter.

Will test it and update. Thanks